src/profile/ProfileForm.tsx

```diff
@@ -37,585 +23,315 @@ interface ChangedUser {
+                <Field
+                  errors={props.context.errors}
+                  type="password"
+                  name="cur_password"
+                  ref={props.context.register({
+                    validate: (password: string) => {
+                      if (
+                        props.user !== undefined &&
+                        props.context.getValues().password &&
+                        !props.context.getValues().cur_password
+                      ) {
+                        return "USER.PASSWORD.CURRENT_REQUIRED";
+                      }
+                    },
+                  })}
+                />
+              </Col>
+            </Form.Group>
+          </>
+        ))}
+      <Form.Group as={Form.Row} controlId="password">
         <Form.Label column sm={2}>
-          New password
-          <OverlayTrigger placement="right" overlay={password_info}>
-            <span>(i)</span>
-          </OverlayTrigger>
+          New Password:
         </Form.Label>
-        <Col sm={10}>
-          <Form.Row>
-            <Col sm="auto">
-              <Button
-                variant="secondary"
-                onClick={() => {
-                  dispatch({
-                    type: "set data",
-                    data: { password: nanoid() },
-                    errors: { status_code: 0, password: [] },
-                  });
-                  setShowPassword(true);
-                }}
-              >
-                Generate
-              </Button>
-            </Col>
-            <Col>
-              <InputGroup>
-                <Form.Control
-                  type={showPassword ? "text" : "password"}
-                  name="password"
-                  onChange={onChange}
-                  isInvalid={
-                    state.errors.password && state.errors.password.length > 0
-                  }
-                  value={state.changedUser?.password ?? ""}
-                />
-                <InputGroup.Append>
-                  <Button
-                    variant={showPassword ? "outline-primary" : "outline-secondary"}
-                    onClick={() => {
-                      setShowPassword((s) => !s);
-                    }}
-                  >
-                    <FontAwesomeIcon icon={showPassword ? faEyeSlash : faEye} />
-                  </Button>
-                </InputGroup.Append>
-              </InputGroup>
-              <Form.Control
-                type="hidden"
-                isInvalid={
-                  state.errors.password && state.errors.password.length > 0
+        <Col sm={2}>
+          <Button
+            className="w-100"
+            onClick={() => {
+              props.context.setValue("password", nanoid());
+              // set the password visible so you can see what you get
+              setShowPassword(true);
+            }}
+          >
+            Generate
+          </Button>
+        </Col>
+        <Col sm={7}>
```

Comment:
Why is the generate password button separated into its own column?

Reply:
Just realized what you did with the nested row thing.
https://github.com/UWmathNEWS/slugline-web/pull/13/commits/f0dfc74467856918a8f3bbf4632d82efb7a04b51